### PR TITLE
demo/fallback.html

### LINKED ISSUE #44: demo/fallback.html
Prepare demo to showcase a fallback UI while loading items

### PR DESCRIPTION
**Pull request by valdrinkoshi**
Fixes #44 by implemeting an example with fallback UI elements
Moved `<contact-element>` into its own file, and added a `sortable` attribute which will display render count and up/down buttons.

##  fallback.html demo

fallback

## bugfixes

While implementing the demo, I found and fixed 2 bugs:
- the value of `last` provided in `rangechange` was offset of -1 
- there might be a race condition when measuring children: measurement is done 1 microtask after `_render()`, and `_render()` is throttled of a microtask, so it might be that 2 consecutive `_render()` might affect the measurement (e.g. the 2nd render hides some children, and the 1st _measureChildren would memoize a 0x0 rect)
screen shot 2018-05-04 at 11 59 37 am
^ to fix this, we debounce _measureCallback

### CODE REVIEW

**Comment by valdrinkoshi** [comment]
@graynorton @domenic PTAL

File: demo/fallback.html [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  </p>
+
+  <virtual-list></virtual-list>
+
+  <script type="module">
+    import '../virtual-list-element.js';
+    import './contacts/contact-element.js';
+
+    const fallbackHeight = 80;
+    const fallbacksCount = Math.ceil(innerHeight / fallbackHeight) + 5;
```
Comment (domenic) [inline demo/fallback.html]: Is this window.innerHeight? Maybe being explicit is better?

A comment explaining why this is calculated this way would be nice.

**Comment by valdrinkoshi** [comment]
Good points, updated! PTAL

File: virtual-repeater.js [hunk 2]
```diff
@@ -222,8 +224,17 @@ export const Repeats = Superclass => class extends Superclass {
    */
   async _measureChildren() {
     if (this._ordered.length > 0) {
+      // Grab the children to be measured. We do this
+      // synchronously as _toMeasure relies on _prevFirst and _prevLast.
       const {indices, children} = this._toMeasure;
+      // We debounce the measurement of a microtask.
+      const id = ++this._measuringId;
       await Promise.resolve();
```
Comment (valdrinkoshi) [inline virtual-repeater.js]: @graynorton do you have suggestions on how to "justify" this microtask?
I was thinking this was added to give time to classes/styles to be applied, but by this time we already did call _newChild and _updateChild, so the user has had already the time to update styles. Also, getBoundingClientRect() forces relayout.

Comment (valdrinkoshi) [reply virtual-repeater.js]: Actually, this helps play nice with browser timing and avoids forcing layout, while still being executed before the next frame. Here a comparison of the demo with & without microtask wait:

 sync execution | await microtask 
---| ---
no-wait |  wait


So, we keep it!

**Pull request merged**